iothub/device/tests/HsmAuthentication/ModuleAuthenticationWithHsmTest.cs

```diff
@@ -33,16 +33,44 @@ public async Task TestSafeCreateNewToken_ShouldReturnSasToken()
             string sasToken = await moduleAuthenticationWithHsm.GetTokenAsync(this.iotHub);
             SharedAccessSignature token = SharedAccessSignature.Parse(iotHub, sasToken);
 
-            string audience = WebUtility.UrlEncode(string.Format(CultureInfo.InvariantCulture, "{0}/devices/{1}/modules/{2}",
```

Comment:
This is different from the prod code yet it works. Can you please add an international domain name (IDN) for the hub. Do we even support that? (If not, maybe we should not UrlEncode the whole string.)

Reply:
It is not different, it is the same. Just that encoding is done in 2 different places with the IoTHub/Prod code (which is probably why it went unnoticed). I just moved it to be in the same place, so that there is no confusion.